orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/KatoService.groovy

```diff
@@ -48,4 +55,22 @@ class KatoService {
 
     return Observable.from(cloudDriverTaskStatusService.lookupTask(id))
   }
+
+  /**
+   * Helper method for generating deterministic client request IDs.
+   *
+   * The callSiteId should be unique within the execution of a Stage. Since operations are
+   * requested from a Stage Task, it's common that there is only one call-site within the Task;
+   * in this case just use the Task class' simpleName.
+   *
+   * It is VERY IMPORTANT that callSiteIds are constant values between invocations, and are
+   * unique from other callSiteIds within the same Stage.
+   *
+   * A Stage's startTime is included to support the case where a Stage has been restarted. In
+   * this case, we would want to treat an operation request as different from a prior request.
+   */
+  @Nonnull
+  static String requestId(@Nonnull Stage stage, @Nonnull String callSiteId) {
+    return "${stage.getId()}:${stage.getStartTime()}:$callSiteId".toString()
+  }
```

Comment:
Would like scrutiny of this: Is my logic sound? Are there cases where this requestId generation strategy would cause bugs?

Reply:
I can 100% guarantee this will cause no bugs

Reply:
1000% now

Reply:
Don’t we still want a `stageId` and `stageStartTime` in here?

Reply:
I'm confused, why don't I see this change in the "Files changed" tab there?

Reply:
@robfletcher this PR was originally 107 files with an alternative implementation for generating the hash ... it's subsequently been trimmed down to 1 file which confused github.

Reply:
Ugh, derp. I didn't mean to merge. Will followup with the stage stuff. :(

Reply:
https://github.com/spinnaker/orca/pull/3080